RevenueCatUI/Templates/Components/CornerBorder.swift

```diff
@@ -0,0 +1,301 @@
+        let bottomRight: CGFloat?
+
+        init(topLeft: Double? = nil, topRight: Double? = nil, bottomLeft: Double? = nil, bottomRight: Double? = nil) {
+            self.topLeft = topLeft.flatMap { CGFloat($0) }
+            self.topRight = topRight.flatMap { CGFloat($0) }
+            self.bottomLeft = bottomLeft.flatMap { CGFloat($0) }
+            self.bottomRight = bottomRight.flatMap { CGFloat($0) }
+        }
+
+    }
+
+    var border: BorderInfo?
+    var radiuses: RaidusInfo?
+
+    func body(content: Content) -> some View {
+        content
+            .conditionalClipShape(topLeft: self.radiuses?.topLeft,
+                                  topRight: self.radiuses?.topRight,
+                                  bottomLeft: self.radiuses?.bottomLeft,
+                                  bottomRight: self.radiuses?.bottomRight)
+            .conditionalOverlay(color: self.border?.color,
+                                width: self.border?.width,
+                                topLeft: self.radiuses?.topLeft,
+                                topRight: self.radiuses?.topRight,
+                                bottomLeft: self.radiuses?.bottomLeft,
+                                bottomRight: self.radiuses?.bottomRight)
+    }
+}
+
+// Helper extensions to conditionally apply clipShape and overlay without AnyView
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+extension View {
+
+    func conditionalClipShape(
+        topLeft: CGFloat?,
+        topRight: CGFloat?,
+        bottomLeft: CGFloat?,
+        bottomRight: CGFloat?
+    ) -> some View {
+        Group {
+            if let topLeft = topLeft,
+                let topRight = topRight,
+                let bottomLeft = bottomLeft,
+                let bottomRight = bottomRight,
+                topLeft > 0 || topRight > 0 || bottomLeft > 0 || bottomRight > 0 {
+                    self
+                        .applyIf(topLeft > 0) {
+                            $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.topLeft]))
+                        }
+                        .applyIf(topRight > 0) {
+                            $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.topRight]))
+                        }
+                        .applyIf(bottomLeft > 0) {
+                            $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.bottomLeft]))
+                        }
+                        .applyIf(bottomRight > 0) {
+                            $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.bottomRight]))
+                        }
+            } else {
+                self
+            }
+        }
+    }
+
+    // swiftlint:disable:next function_parameter_count
+    func conditionalOverlay(
+        color: Color?,
+        width: CGFloat?,
+        topLeft: CGFloat?,
+        topRight: CGFloat?,
+        bottomLeft: CGFloat?,
+        bottomRight: CGFloat?
```

Comment:
Not sure if this makes sense, but we could also make these non-nil and default to 0? That way you can apply rounding to some corners, without having to specify all corners.